internal/node/node.go

```diff
@@ -335,6 +335,7 @@ func (n *Node) shutdown() {
 		n.ncint.Close()
 		n.natsint.Shutdown()
 		n.natsint.WaitForShutdown()
+		_ = n.telemetry.providerShutdown(n.ctx)
```

Comment:
I would create a `Shutdown()` function on `Telemetry` itself and call `t.providerShutdown()` from there.